shared/util.go

```diff
@@ -71,9 +72,22 @@ func RunCommand(name string, arg ...string) error {
 // and redirecting the process's stdout and stderr to the real stdout and stderr
 // respectively.
 func RunScript(content string) error {
-	cmd := exec.Command("sh")
+	fd, err := unix.MemfdCreate("tmp", 0)
+	if err != nil {
+		return err
+	}
+	defer unix.Close(fd)
+
+	_, err = unix.Write(int(fd), []byte(content))
+	if err != nil {
+		return err
+	}
+
+	fdPath := fmt.Sprintf("/proc/self/fd/%d", fd)
 
-	cmd.Stdin = bytes.NewBufferString(content)
+	cmd := exec.Command(fdPath)
+
+	cmd.Stdin = os.Stdin
 	cmd.Stdout = os.Stdout
 	cmd.Stderr = os.Stderr
```

Comment:
With this change, we don't do anything fancy with exec.Command anymore so can move to our standard `shared.RunCommand(fdPath)` instead.

Reply:
Good call